wavefront_lambda/wavefront_lambda/__init__.py

```diff
@@ -0,0 +1,112 @@
+from wavefront_pyformance.wavefront_reporter import WavefrontDirectReporter
+from pyformance import MetricsRegistry
+import os
+from datetime import datetime
+from wavefront_pyformance import delta
+
+is_cold_start = True
+reg = None
+
+
+def wrapper(func):
+    """
+    Returns the Wavefront Lambda wrapper. The wrapper collects aws lambda's
+    standard metrics and reports it directly to the set wavefront url. It
+    requires the following Environmental variables to be set:
+    1.WAVEFRONT_URL : your corp wavefront url
+    2.WAVEFRONT_API_TOKEN : The users API authentication token
+    3.IS_REPORT_STANDARD_METRICS : set to false to not report standard lambda
+                                   metrics directly to wavefront.
+
+    """
+    def call_lambda_with_standard_metrics(wf_reporter, *args, **kwargs):
+        # Set cold start counter
+        global is_cold_start
+        if is_cold_start:
+            aws_cold_starts_counter = delta.delta_counter(reg, "coldstart")
+            aws_cold_starts_counter.inc()
+            aws_cold_starts_normal_counter = reg.counter("coldstart-normalcounter")
```

Comment:
Let's remove the `-normalcounter` suffix. Maybe `coldstarts_raw`?

Also wondering if this should be a gauge, is it possible for this counter to ever be > 1? i.e for two cold starts resulting in the same counter instance being used?

Reply:
cold start can only be True/False where True => 1 , If a 2nd container is spun up as a result of concurrent invocations and hence the 2nd cold start , then the 2nd container will use a new registry, so above scenario should not come up.